storage/storagepb/convert.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagepb
+
+import (
+	"fmt"
+
+	"github.com/google/trillian/merkle/smt"
+	"github.com/google/trillian/storage/tree"
+)
+
+// Unmarshal converts the given SubtreeProto to a Merkle tree tile.
+func Unmarshal(sp *SubtreeProto) (smt.Tile, error) {
```

Comment:
Love the Unmarshal semantics. Typically, however the Unmarshal signatures look more like:
```suggestion
func Unmarshal(sp *SubtreeProto, tile *smt.Tile) error {
```
eg. https://golang.org/pkg/encoding/json/#Unmarshal

On the contrary though, those type of signatures are only relevant for general functions where they are doing reflection on the destination type - so disregard. 

The second question about the signature is on the pros/cons of returning a pointer or not. 
Pointer receivers are standard, and allow returning `nil` on error conditions, but have the downside of making escape analysis slightly harder. Is there are reason you've chosen an immediate return type?